repo/blob/azure/azure_versioned_test.go

```diff
@@ -0,0 +1,219 @@
+package azure_test
+
+import (
+	"context"
+	"crypto/rand"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/require"
+
+	"github.com/kopia/kopia/internal/clock"
+	"github.com/kopia/kopia/internal/gather"
+	"github.com/kopia/kopia/internal/testlogging"
+	"github.com/kopia/kopia/internal/testutil"
+	"github.com/kopia/kopia/repo/blob"
+	"github.com/kopia/kopia/repo/blob/azure"
+	"github.com/kopia/kopia/repo/format"
+)
+
+func TestGetBlobVersions(t *testing.T) {
```

Comment:
@KastenMike This test and the one below are failing in the "test/providers" job. The likely cause is that the buckets used in the test do not have the required setup, that is version-level immutability enabled.

Reply:
Did you re-run the tests? I cannot see run history & you didn't link a new run. Now it doesn't check for version-level immutability

Reply:
OK I see it in Actions. Someone should change enable versioning on the related storage account for it to work

Reply:
In the process whoever sets it could enable version-level immutability on the container since it seems the SDK doesn't support it

Reply:
@jkowalski is it something you'd be able to enable?

Reply:
Mike: this requires a little more than that.

These 2 tests need to run in a different setup. That means:
- setting up a **separate** storage account with version-level immutability enabled (as you mentioned).
- create a new bucket with version-level immutability, or have the test create it with the desired settings.
- these tests need to use a separate 'credentials setting' from the environment. The actual credentials may be the same, but the storage account and bucket would be different.
- the GH environment needs to be updated with these new cred information.

Reply:
Why does it require a separate storage account? you could simply have the same one running new + existing tests but with 2 new settings changed

Reply:
@KastenMike For what I understand, the storage account currently used in the tests does not have immutability enabled. I thought that version immutability needed to be enabled at the storage account level. This is what I recall but I could be mistaken. I'll double check.

If no separate storage account is needed, the it is a matter of simply creating a separate container with the appropriate settings.

PTAL at https://github.com/kopia/kopia/compare/azure-immutability-setup

This contains the setup code for the storage account and the container. PLMK if that makes sense, if there's anything missing or that needs to be changed.

Thanks.

Reply:
Setting `ImmutableStorageWithVersioning` to true via CLI is automatically making versioning be enabled too? In the UI they make you do 2 separate actions. A policy isn’t needed on the container~ only versioning + ImmutableStorageWithVersioning
If the intention would be to run it repeatedly like the `createContainer` currently then it needs checks for "already exists" errors also on the SA & container

Reply:
Behavior observed using the console: attempting to enable "version immutability" in the container requires enabling "versioning" in the storage account. It wasn't clear at that moment what the implication was for existing buckets, so we created a separate storage account and bucket.

See inline comment as well.